src/Sarif.Converters/FileInfoFactory.cs

```diff
@@ -96,6 +93,13 @@ private void AddFile(PhysicalLocation physicalLocation)
 
             Uri uri = physicalLocation.FileLocation.Uri;
             string key = UriHelper.MakeValidUri(uri.OriginalString);
+
+            string uriBaseId = physicalLocation.FileLocation.UriBaseId;
+            if (!string.IsNullOrEmpty(uriBaseId))
+            {
+                key = "#" + uriBaseId + "#" + key;
```

Comment:
>"#" + uriBaseId + "#" + key; [](start = 22, length = 28)

This is ok if the SDK is consistent about it, and I suppose it's the safest course.

The spec doesn't require the `#<uriBaseId>` prefix _unless_ the dictionary needs to contain entries for two files with the same `uri` but different `uriBaseIds`. You can't actually determine that without looking at every `fileLocation` object in the log file.

So yes, I think this is the right thing to do. Just thinking out loud.